dpctl/program/_program.pxd

```diff
@@ -33,26 +33,26 @@ cdef class SyclKernel:
         kernel.
     '''
     cdef DPCTLSyclKernelRef _kernel_ref
-    cdef const char *_function_name
+    cdef str _function_name
```

Comment:
Why the change? It is for my education about Cython mostly.

Reply:
Mostly to delegate copying and memory management to Python, which simplifies the code. No other reason.